packages/jmix-react-web/src/crud/editor/ui-callbacks/useSubmitCallback.ts

```diff
@@ -2,45 +2,49 @@ import {useCallback} from "react";
 import {
   addIdIfExistingEntity,
   EntityInstance,
-  findEntityMetadata,
-  GraphQLMutationFn,
   Metadata,
   toIdString,
   useMetadata,
-  unCapitalizeFirst,
-  MetaClassInfo
 } from "@haulmont/jmix-react-core";
-import { PersistEntityCallbacks, persistEntity } from "../util/persistEntity";
+import { ApolloError, FetchResult } from "@apollo/client";
+import {GraphQLError} from 'graphql';
 
-export interface SubmitCallbackHookOptions<TEntity, TData, TMutationVars> {
-  executeUpsertMutation: GraphQLMutationFn<TData, TMutationVars>;
-  updateResultName: string;
-  listQueryName: string;
+export interface SubmitCallbackCallbacks<TEntity, TData> {
+  onCommit?: (entityInstance: EntityInstance<TEntity>) => void;
+  onCreate?: (data?: TData | null) => void;
+  onEdit?: (data?: TData | null) => void;
+  onError?: (errors?: readonly GraphQLError[]) => void;
+  onApolloError?: (error: Error | ApolloError) => void;
```

Comment:
We need to have better names.

`onCommit` - it should be clear that this callback is responsible for doing something with the data, e.g. persisting, or updating client state. (maybe `onSave`?)

`onCreate`, `onEdit` - should be clear that they are responsible for updating UI (e.g. displaying a message and going to previous screen).

`onError`, `onApolloError` - should be clear what is the difference between the two